src/amp-story-player/amp-story-player-impl.js

```diff
@@ -551,20 +551,38 @@ export class AmpStoryPlayer {
       return;
     }
 
+    let firstStoryLoadedPromise;
     for (let idx = 0; idx < this.stories_.length && idx < MAX_IFRAMES; idx++) {
       const story = this.stories_[idx];
       const iframeIdx = story[IFRAME_IDX];
       const iframe = this.iframes_[iframeIdx];
-      this.layoutIframe_(
-        story,
-        iframe,
-        idx === 0 ? VisibilityState.VISIBLE : VisibilityState.PRERENDER
-      );
+      if (idx === 0) {
+        this.layoutIframe_(story, iframe, VisibilityState.VISIBLE);
+        firstStoryLoadedPromise = this.waitForStoryToLoadPromise_(iframeIdx);
+      } else {
+        // Only layout next story after first one is loaded.
+        firstStoryLoadedPromise.then(() => {
```

Comment:
What happens if you swipe next before the first story is loaded?

Reply:
Good point, addressed in latest commit!